keps/sig-cli/0031-kustomize-integration.md

```diff
@@ -32,122 +33,210 @@ superseded-by:
+Solve common kubectl user friction (such as those defined in the Motivation section) by publishing Kustomize
+functionality from kubectl to complement commands which are targeted at declarative workflows.
 
+- Complement commands targeted at declarative workflows: `kubectl apply`, `kubectl diff`
+- Complement the *server-side apply* initiative targeted at improving declarative workflows.
 
-### Non-Goals
-- provide an editing functionality of kustomization.yaml from kubectl
-- further integration with other kubectl flags
-
-## Kustomize Introduction
-
-Kustomize has following subcommands:
-- build
-- edit, edit also has subcommands
-   - Set 
-       - imagetag
-       - namespace
-       - nameprefix
-   - Add
-       - base
-       - resource
-       - patch
-       - label
-       - annotation
-       - configmap
-- config
-- version
-- help
-
-`edit` and `build` are most commonly used subcommands. 
-
-`edit` is to modify the fields in `kustomization.yaml`. A `kustomization.yaml` includes configurations that are consumed by Kustomize. Here is an example of `kustomization.yaml` file.
-
-`build` is to perform a set of pre-processing transformations on the resources inside one kustomization. Those transformations include:
-- Get objects from the base
-- Apply patches
-- Add name prefix to all resources
-- Add common label and annotation to all resources
-- Replace imageTag is specified
-- Update objects’ names where they are referenced
-- Resolve variables and substitute them
+User friction solved through capabilities such as:
 
-```
-apiVersion: v1beta1
-kind: Kustomization
-namePrefix: alices-
+- Generating Resource Config for Resources with data frequently populated from other sources - ConfigMap and Secret
+- Performing common cross-cutting transformations intended to be applied across Resource Configs - e.g.
+ name prefixing / suffixing, labeling, annotating, namespacing, imageTag setting.
+- Performing flexible targeted transformations intended to be applied to specific Resource Configs - e.g.
+ strategic patch, json patch
+- Composing and layering Resource Config and schema-aware transformations
+- Facilitating rolling updates to Resources such as ConfigMaps and Secrets
+- Facilitating creation of Resources that require the creation to be ordered - e.g. namespaces, crds
 
-commonAnnotations:
- oncallPager: 800-555-1212
+### Non-Goals
 
-configMapGenerator:
-- name: myJavaServerEnvVars
- literals:   
- - JAVA_HOME=/opt/java/jdk
- - JAVA_TOOL_OPTIONS=-agentlib:hprof
+- Exposing all imperative kubectl commands as declarative options
+- Exposing all kubectl generators and schema-aware transformations
+- Providing simpler alternatives to the APIs for declaring Resource Config (e.g. a simple way to create deployments)
+- Providing a templating mechanism (e.g. for generating Resource Config)
+- Publishing with kubectl any other Kustomize sub-commands besides `build`
```

Comment:
That's the one I had biggest issues with in the original PR. I'd like to narrow the list of those commands to a specific set. I don't want to end up with nasty commands like `kubectl kustomize edit myresource` which will confuse people with `kubectl edit myresource`, etc.

Reply:
Can we just keep this as a non-goal and not do it?

Reply:
Do we need to include it, now that we agree it won't be?